gridmet_split_script.py

```diff
@@ -0,0 +1,219 @@
+
+
+def get_gridmet_datasets(variable, start_date, end_date, polygon_as_bbox = None, lon_min = None, lat_min = None, lon_max = None, lat_max = None):
+
+    """
+    :param str/list var_short: data variable short name or list of data variable short names. Must be one of the following: ['tmmx', 'tmmn', 'pr', 'srad', 'vs', 'rmax', 'rmin', 'sph']
+    :param str start_date: Start date of data collection yyyy-mm-dd
+    :param str end_date: End date of data collection yyyy-mm-dd
+    :param gdf.total_bounds polygon_bbox: total bounds of a geodataframe. Ex: geodataframe.total_bounds. If not known, None
+    :param str lon_min: bbox of aoi longitude min. None if polygon_bbox is given (i.e. polygon_bbox != None)
+    :param str lat_min: bbox of aoi latitude min. None if polygon_bbox is given (i.e. polygon_bbox != None)
+    :param str lon_max: bbox of aoi longitude max. None if polygon_bbox is given (i.e. polygon_bbox != None)
+    :param str lat_max: bbox of aoi latitude max. None if polygon_bbox is given (i.e. polygon_bbox != None)
+    :return: dictionary of xarray stored by variable name
+    """
+
+    ## check/define bounds for data slicing
+    if polygon_as_bbox is not None:
+        if isinstance(polygon_as_bbox, geopandas.geodataframe.GeoDataFrame):
+            print('polygon is geodataframe')
+            print(polygon_as_bbox.total_bounds)
+            pass
+        elif os.path.isfile(polygon_as_bbox):
+            print('polygon is path to shapefile')
+            polygon_as_bbox = gpd.read_file(polygon_as_bbox)
+            print(polygon_as_bbox.total_bounds)
+        else:
+             raise TypeError('polygon_as_bbox should str path to shapefile or geodataframe')
+
+        if polygon_as_bbox.crs != 'EPSG:4326':
+            polygon_as_bbox = polygon_as_bbox.to_crs('EPSG:4326')
+            print('geodataframe crs changed to EPSG:4326')
+
+        lon_min, lat_min, lon_max, lat_max = polygon_as_bbox.total_bounds
+    else:
+        pass
+
+    ## if only 1 var (short) is provided, change to list
+    if not isinstance(variable, list):
+        variable = [variable]
+
+    ## Initiate list of empty datasets
+    xarray_dict = dict()
+
+    ## Loop through variables and pull data + place in dict
+    for var in variable:
+        ## Pulling data
+        # source: http://thredds.northwestknowledge.net:8080/thredds/reacch_climate_MET_aggregated_catalog.html
+        url = f'http://thredds.northwestknowledge.net:8080/thredds/dodsC/agg_met_{var}_1979_CurrentYear_CONUS.nc'
+        # call data from url
+        start = time.perf_counter()
+        ds = xr.open_dataset(url + '#fillmismatch')
+
+        ## Subset to timeframe and bbox of interest:
+        ds_subset = ds.sel(day = slice(start_date, end_date),
+                           lon = slice(lon_min, lon_max),
+                           lat = slice(lat_max, lat_min))
+        end = time.perf_counter()
+        print(f'finish in {round(start - end, 2)} seconds')
```

Comment:
suggest this so the time elapsed is positive
```suggestion
        print(f'finish in {round(end - start, 2)} seconds')
```